dataprofiler/profilers/profile_builder.py

```diff
@@ -365,6 +390,38 @@ def __init__(self, data, samples_per_update=None, min_true_samples=None,
         if isinstance(data, data_readers.text_data.TextData):
             raise TypeError("Cannot provide TextData object to Profiler")
 
+        # assign data labeler
+        use_data_labeler = True
+        data_labeler_dirpath = None
+        self.data_labeler = None
+        structured_options = None
+
+        if profiler_options and profiler_options.structured_options:
+            structured_options = profiler_options.structured_options
+
+        if structured_options and isinstance(
+                structured_options, StructuredOptions):
+            data_labeler_options = structured_options.data_labeler
+
+        if data_labeler_options and isinstance(
+                data_labeler_options, DataLabelerOptions):
+            use_data_labeler = data_labeler_options.is_enabled
```

Comment:
If we receive options and they are `profiler_options` we can assume it has `structured_options` and should not need these checks. However, at the top we guarantee options to be ProfilerOptions by setting it to the class.

Additionally, here we only want to add a default_labeler if none exists in the options

Hence, I think we can simplify to:
```python
# assign data labeler
data_labeler_options = self.options.structured_options.data_labeler
if data_labeler_options.is_enabled and data_labeler_options.data_labeler_object is None:
    data_labeler = DataLabeler(
        labeler_type='structured',
        dirpath=data_labeler_options.data_labeler_dirpath,
        load_options=None)
    self.options.set({'data_labeler.data_labeler_object': data_labeler})
```

Reply:
I tried but it has some error. But I think it can definitely be simplified later.